pyiceberg/table/__init__.py

```diff
@@ -231,9 +233,13 @@ def __enter__(self) -> Transaction:
         """Start a transaction to update the table."""
         return self
 
-    def __exit__(self, _: Any, value: Any, traceback: Any) -> None:
-        """Close and commit the transaction."""
-        self.commit_transaction()
+    def __exit__(
+        self, exctype: Optional[Type[BaseException]], excinst: Optional[BaseException], exctb: Optional[TracebackType]
+    ) -> None:
+        """Close and commit the transaction, or handle exceptions."""
+        # Only commit the full transaction, if there is no exception in all updates on the chain
+        if exctb is None:
```

Comment:
Since [python doc](https://docs.python.org/3/reference/datamodel.html#object.__exit__) mention that

> If the context was exited without an exception, all three arguments will be [None](https://docs.python.org/3/library/constants.html#None).

Shall we check all three parameter is `None` here?

Reply:
ah, you are right, thanks for pointing this out.